### PR TITLE
Add 2023 day 23 solution and input

### PR DESCRIPTION
**Pull request by stewSquared**

### CODE REVIEW

File: 2023/input/day23 [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#.#####################
```
Comment (bishabosha) [inline 2023/input/day23:1]: Apparently these inputs shouldn't be shared (look it up), so now we will no longer accept them :)

Comment (stewSquared) [reply 2023/input/day23:1]: I know! These aren't my inputs. They're the example :)

Comment (bishabosha) [reply 2023/input/day23:1]: well I think even the sample input is still protected

**Comment by bishabosha** [comment]
Lovely solutions all around, thank you again!

**Pull request merged**